tests/lcp/test_collection.py

```diff
@@ -0,0 +1,516 @@
+                configuration.max_copiable_pages = LCPServerConfiguration.DEFAULT_MAX_COPIABLE_PAGES
+
+                # Act
+                loan = lcp_api.checkout(patron, 'pin', license_pool, 'internal format')
+
+                # Assert
+                eq_(loan.collection_id, self._lcp_collection.id)
+                eq_(loan.collection(self._db), self._lcp_collection)
+                eq_(loan.license_pool(self._db), license_pool)
+                eq_(loan.data_source_name, data_source_name)
+                eq_(loan.identifier_type, license_pool.identifier.type)
+                eq_(loan.external_identifier, lcp_license['id'])
+                eq_(loan.start_date, today)
+                eq_(loan.end_date, expires)
+
+    @freeze_time("2020-01-01 00:00:00")
+    def test_fulfil(self):
+        # Arrange
+        lcp_api = LCPAPI(self._db, self._lcp_collection)
+        patron = self._patron()
+        days = self._lcp_collection.default_loan_period(patron.library)
+        today = datetime.datetime.utcnow()
+        expires = today + datetime.timedelta(days=days)
+        data_source = DataSource.lookup(self._db, DataSource.LCP, autocreate=True)
+        data_source_name = data_source.name
+        license_pool = self._licensepool(
+            edition=None, data_source_name=data_source_name, collection=self._lcp_collection)
+        lcp_license = json.loads(fixtures.LCPSERVER_LICENSE)
+        lcp_server_mock = create_autospec(spec=LCPServer)
+        lcp_server_mock.generate_license = MagicMock(return_value=lcp_license)
+
+        with self._configuration_factory.create(
+                self._configuration_storage, self._db, LCPServerConfiguration) as configuration:
+            with patch('api.lcp.collection.LCPServer') as lcp_server_constructor:
+                lcp_server_constructor.return_value = lcp_server_mock
+
+                configuration.lcpserver_url = fixtures.LCPSERVER_URL
+                configuration.lcpserver_user = fixtures.LCPSERVER_USER
+                configuration.lcpserver_password = fixtures.LCPSERVER_PASSWORD
+                configuration.lcpserver_input_directory = fixtures.LCPSERVER_INPUT_DIRECTORY
+
+                configuration.provider_name = fixtures.PROVIDER_NAME
+                configuration.passphrase_hint = fixtures.TEXT_HINT
+                configuration.encryption_algorithm = LCPServerConfiguration.DEFAULT_ENCRYPTION_ALGORITHM
+                configuration.max_printable_pages = LCPServerConfiguration.DEFAULT_MAX_PRINTABLE_PAGES
+                configuration.max_copiable_pages = LCPServerConfiguration.DEFAULT_MAX_COPIABLE_PAGES
+
+                # Act
+                license_pool.loan_to(patron, start=today, end=expires, external_identifier=lcp_license['id'])
+                fulfilment_info = lcp_api.fulfill(patron, 'pin', license_pool, 'internal format')
+
+                # Assert
+                eq_(fulfilment_info.collection_id, self._lcp_collection.id)
```

Comment:
Also verify that `fulfilment_info` is an `LCPFulfilmentInfo` rather than some other kind.

Reply:
Added an assertion